OpenRA.Mods.RA/Activities/Infiltrate.cs

```diff
@@ -50,11 +48,19 @@ protected override void OnInside(Actor self)
 
 			var exp = self.Owner.PlayerActor.TraitOrDefault<PlayerExperience>();
 			if (exp != null)
-				exp.GiveExperience(experience);
+				exp.GiveExperience(trait.Info.PlayerExperience);
 
-			if (!string.IsNullOrEmpty(notification))
+			if (!string.IsNullOrEmpty(trait.Info.Notification))
 				Game.Sound.PlayNotification(self.World.Map.Rules, self.Owner, "Speech",
-					notification, self.Owner.Faction.InternalName);
+					trait.Info.Notification, self.Owner.Faction.InternalName);
+		}
+
+		public override Activity Tick(Actor self)
+		{
+			if (trait.IsTraitDisabled)
+				return CanceledTick(self);
```

Comment:
This case is a classic example of how you can't assume that cancelling an activity will immediately advance to the next activity.  Many activities (especially those relating to movement) need several ticks to bring the actor back to a self-consistent state (e.g. standing at one of the allowed idle cell positions) before it can move on, and bugs like #9404 happen when code ignores this requirement.